### ISSUE TITLE
Cannot create Parse Role after "fix: return correct response when revert is used in beforeSave #7839"

### ISSUE THREAD

**Issue by AlbinIzi** [parse-community/parse-server#8317]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

The PR https://github.com/parse-community/parse-server/pull/7839 introduce a regression. We cannot create a Parse.Role anymore if we have a beforeSave on Parse.Role.
We get an error: TypeError: Tried to create an ACL with an invalid permission type.

### Steps to reproduce

Add this test in spec/CloudCode.spec.js:

```
  it('create role with name and ACL and a beforeSave', async () => {
    Parse.Cloud.beforeSave(Parse.Role, ({ object }) => {
      // do nothing, it's just to reproduce the bug, we must have a beforeSave
      return object;
    });

    const obj = new Parse.Role('TestRole', new Parse.ACL({ '*': { read: true, write: true } }));
    await obj.save();

    expect(obj.getACL()).toEqual(new Parse.ACL({ '*': { read: true, write: true } }));
    expect(obj.get('name')).toEqual('TestRole');
    await obj.fetch();

    expect(obj.getACL()).toEqual(new Parse.ACL({ '*': { read: true, write: true } }));
    expect(obj.get('name')).toEqual('TestRole');
  });
```

### Actual Outcome

>   Message:
>     **TypeError: Tried to create an ACL with an invalid permission type.**
>   Stack:
>         at <Jasmine>
>         at new ParseACL (/Users/albin/iziwork/parse-server/node_modules/parse/lib/node/ParseACL.js:69:21)
>         at ParseRole._handleSaveResponse (/Users/albin/iziwork/parse-server/node_modules/parse/lib/node/ParseObject.js:614:25)
>         at /Users/albin/iziwork/parse-server/node_modules/parse/lib/node/ParseObject.js:3183:22
>         at processTicksAndRejections (node:internal/process/task_queues:96:5)

### Expected Outcome

The created role with the right name and ACL.

### Environment

Server
- Parse Server version: `5.3.3`
- Operating system: `MacOS`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `5.2.1`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `JavaScript`
- SDK version: `3.4.2`

**Comment by AlbinIzi** [comment]
From my understanding, with PR [7839](https://github.com/parse-community/parse-server/pull/7839), the `buildParseObjects` function is now called after the save.
The problem is that `buildParseObjects` function call `set` on `Parse.Role` that call `validate` on `Parse.Role` which failing (ParseError: A role's name can only be set before it has been saved) so pendingOps are not set on the new created Parse.Role object.
The result is that we have a difference between pendingOps before the beforeSave and after the save so the name and ACL fields are marked as deleted: `{ __op: 'Delete' }` which is not a valid ACL value.

**Comment by AlbinIzi** [comment]
I don't know the impacts but since `buildParseObjects` was not called before and because validation have already been done, we could add an optional `options` parameter to `buildParseObjects` function (default value: `{ ignoreValidation: false }`) to use it when `updatedObject.set(sanitized)` is called.
In `_updateResponseWithData` the buildParseObjects will be called with `{ ignoreValidation: true }` to disable validation after the save.
What do you think ?

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.10](https://github.com/parse-community/parse-server/releases/tag/6.0.0-alpha.10)

**Comment by AlbinIzi** [comment]
Thanks ! Is it possible to fix this issue in 5.3.4 please ?

**Comment by mtrezza** [comment]
See my answer [here](https://github.com/parse-community/parse-server/pull/8320#issuecomment-1320866690)

**Comment by mtrezza** [comment]
See my answer [here](https://github.com/parse-community/parse-server/pull/8320#issuecomment-1320866690)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/parse-server/releases/tag/6.0.0)
